File: 0118-pascals-triangle/0118-pascals-triangle.go
func generate_next_row(prevRow []int) []int {
    nextRow := make([]int, 0);
    nextRow = append(nextRow, 1);
    for i := range prevRow[1:] {
        nextRow = append(nextRow, prevRow[i] + prevRow[i + 1]);
    }
    nextRow = append(nextRow, 1);
    return nextRow;
}

func generate(numRows int) [][]int {
    res := make([][]int, 0);
    res = append(res, []int{1});
    for i := 1; i < numRows; i += 1 {
        res = append(res, generate_next_row(res[i - 1]));
    }
    return res;
}

File: 0118-pascals-triangle/0118-pascals-triangle.rb
def generate_next_rows(prev_rows)
    new_rows = Array[1];
    for i in 1...prev_rows.size
        new_rows.append(prev_rows[i] + prev_rows[i - 1]);
    end
    new_rows.append(1);
    return new_rows;
end

# @param {Integer} num_rows
# @return {Integer[][]}
def generate(num_rows)
    res = Array[Array[1]];
    for i in 1...num_rows
        res.append(generate_next_rows(res[i - 1]));
    end
    return res;
end